src/cliffordsynthesis/encoding/SATEncoder.cpp

```diff
@@ -63,15 +67,25 @@
   if (config.useMultiGateEncoding) {
     gateEncoder = std::make_shared<MultiGateEncoder>(
         N, s, T, tableauEncoder->getVariables(), lb);
+    gateEncoder->createSingleQubitGateVariables(0U);
+    gateEncoder->createTwoQubitGateVariables(0U);
+  } else if (config.useTwoQubitEncoding) {
+    T -= 1;
+    gateEncoder = std::make_shared<TwoQubitEncoder>(
+        N, s, T, tableauEncoder->getVariables(), lb);
+    gateEncoder->createSingleQubitGateVariables(1U);
+    gateEncoder->createTwoQubitGateVariables(1U);
+    gateEncoder->createPauliQubitGateVariables();
   } else {
     gateEncoder = std::make_shared<SingleGateEncoder>(
         N, s, T, tableauEncoder->getVariables(), lb);
+    gateEncoder->createSingleQubitGateVariables(0U);
+    gateEncoder->createTwoQubitGateVariables(0U);
   }
-  gateEncoder->createSingleQubitGateVariables();
-  gateEncoder->createTwoQubitGateVariables();
+
   gateEncoder->encodeGates();
 
-  if (config.useSymmetryBreaking) {
+  if (config.useSymmetryBreaking && !config.useTwoQubitEncoding) {
     gateEncoder->encodeSymmetryBreakingConstraints();
   }
```

Comment:
Will have to be changed once methods are adjusted.
Also, you don't have to check for `useTwoQubitEncoding` if you make the `useSymmetryBreaking` method just not do anything. Then we don't have this strange special case here and don't have to adapt anything if somebody decides to implement symmetry breakers in the future.